### PR TITLE
Dynamic Hotel Search and Retrieval with test UI components

### PR DESCRIPTION
**Pull request by ray-sjsu**
Navigate to `http://localhost:3000/searchtest` to see test page.

All parameters are included, but most are optional. Included are test UI components to change the parameters. We don't actually need all of them, but this makes it easier for testing. Default values are provided. Look at `constants.ts`.

Instructions:
1. Enter Region Search Query. `San Jose` should be already filled out.
2. Click `Find Region` button.
3. Click any region item in the list. After clicking on an item, only one region should show up. Check `Selected Region ID` is filled out and not `N/A`.
4. Scroll down and click `Find Hotels` button. This should be blue if everything is filled out correctly.
5. Click on any hotel item.
6. You should be redirected to  `http://localhost:3000/hotels/{hotelId}?searchParams`

This is a temporary route that bypasses any created ones. This will be later removed with later discussion.
The format should be:
`http://localhost:3000/hotels/{hotelId}?searchParams`

### CODE REVIEW

File: lib/rapid-hotel-api/constants.ts
Comment (Luceium) [inline lib/rapid-hotel-api/constants.ts:1]: Great organization

File: next.config.mjs
Comment (Luceium) [inline next.config.mjs:1]: Might not be the best decision but, it's not to impactful either.

File: lib/DateFunctions.ts
Comment (Luceium) [inline lib/DateFunctions.ts:1]: @alia49 This may be a good place for any date related helper functions you make

File: components/search/Templates-UI/TemplateCheckbox.tsx
Comment (Luceium) [inline components/search/Templates-UI/TemplateCheckbox.tsx:1]: Why was a new UI lib added?

File: app/api/hotels/search/rooms/route.ts
Comment (Luceium) [inline app/api/hotels/search/rooms/route.ts:1]: @alia49 you'll be adding your function call to this file to only return rooms that aren't booked.

**Review by Luceium — APPROVED** [review]
I noticed that the testing paths have `(test)` which won't be included in the URL.
Let's use `test` to make it clear when navigating and testing the page.

Besides that this looks well organized.
Next time, try to make smaller more focused PRs.

All feedback are just NITs so feel free to address and approve after, no need for a second review.

**Review by ryanhtang — APPROVED** [review]

**Comment by ray-sjsu** [comment]
Added room details and a dedicated booking page for a specific room. That page in particular `/api/hotels/{hotelId}/{roomId} needs to be integrated with the payment system and requires more detail like hotel information, address (which is a separate API call).

That will be for a future pull request.

**Review by Luceium — APPROVED** [review]

**Pull request merged**
